app/src/main/java/com/taskruler/MainViewModel.kt

```diff
@@ -31,7 +31,7 @@ class MainViewModel(var activityService : IActivityService = ActivityService())
     var user : User? = null
     internal val NEW_TASK: String = "New Task"
 
-    private lateinit var firestore : FirebaseFirestore
+    private var firestore : FirebaseFirestore
```

Comment:
don't really need lateinit since you initialize variable in the next block anyways

Reply:
Hmmmm, that's when you need lateinit.  Without that, and without assigning this immediately upon declaration, you'd either need to initialize it to null, or it will throw a compile error.